templates/template.rb

```diff
@@ -348,11 +351,10 @@ def locals
         begin
           config = YAML.load_file(File.expand_path("../config.yml", __dir__))
 
-          {
-            nodes: config.fetch("nodes").map { |node| NodeType.new(node) }.sort_by(&:name),
-            tokens: config.fetch("tokens").map { |token| Token.new(token) },
-            flags: config.fetch("flags").map { |flags| Flags.new(flags) }
-          }
+          nodes = config.fetch("nodes").map { |node| NodeType.new(node) }.sort_by(&:name)
+          tokens = config.fetch("tokens").map { |token| Token.new(token) }
+          flags = config.fetch("flags").map { |flags| Flags.new(flags) }
+          binding
```

Comment:
I'm not keen to expose the entire binding to the template files, I'd much rather keep the scope of what they can access limited. Can you revert this change?

Reply:
The only extra thing available in this binding is `config` which is not used anyway (and we could hide it if we care about that one), and having a constant scope of `YARP` which is quite a bit more convenient.
Is that the concern?

I can use `YARP::JAVA_STRING_TYPE` otherwise, but it is really verbose for every place I need the type.

Reply:
Yeah we do that with the other constants, I'd rather keep it consistent.

Reply:
Actually I'd be happy to remove all those `YARP::` prefixes in templates for field classes, they used to not be there and IMO it was more a little bit more readable that way. But I didn't want to do that big change in this PR.

I'll just undo this part, we can always do it later and should be its own PR anyway for consistency.

Reply:
I reverted those changes.